bwscanner/scanner.py

```diff
@@ -29,18 +38,27 @@ def __repr__(self):
 pass_scan = click.make_pass_decorator(ScanInstance)
 
 
-@click.group()
+# FIXME: change all options to take defaults from CTX, ie config file?
+@click.group(context_settings=CTX)
 @click.option('--data-dir', type=click.Path(),
```

Comment:
So, instead of this and the CTX.get() stuff below, I think you can: re-define the entry-point as say `start` and make that a method like:
```
def start():
    config = read_config(os.path.join(DATA_DIR, CONFIG_FILE))
    return cli(default_map=config)
```

See http://click.pocoo.org/5/commands/#overriding-defaults too

Reply:
Hmm, by reading the docs is not clear to me where ``start`` is supposed to be call.
By the "stuff below" you mean all the ``default=`` in the options?

Reply:
You meant something like https://github.com/juga0/bwscanner/commit/388e824afa669061d975ff7a9947d3abb7811579#diff-091411b9c979317818f6582241ab8284?

Reply:
Yeah, that's what I meant. I haven't actually done this config-file-based defaults myself, just basing it off Click's docs. I wouldn't expect you'd have to do that "for k, v in ...default_map" thing either based on what Click *says* but .... I can play with it a bit myself later.